sklearn/isotonic.py

```diff
@@ -57,7 +57,7 @@ def check_increasing(x, y):
     increasing_bool = rho >= 0
 
     # Run Fisher transform to get the rho CI, but handle rho=+/-1
-    if rho not in [-1.0, 1.0]:
+    if (rho not in [-1.0, 1.0]) and (len(x) > 3):
```

Comment:
Python is not C, you don't need the parentheses around boolean conditions. Replace this by:
```py
if rho not in [-1.0, 1.0] and len(x) > 3:
```